server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -65,6 +66,13 @@
  */
 public class DenseVectorFieldMapper extends FieldMapper {
 
+    // Dear reader, this is a weird bug.
+    // If we first load the PanamaVector module from a Painless context, we run into two failures
+    //   1. log4j needs privileged access to get environment variable values to print out the preferred bit size
+    //   2. loopBound proved by the module makes a privileged call `jdk.incubator.vector.VECTOR_ACCESS_OOB_CHECK`
+    //      this is apparently cached as once its loaded, we don't run into issues again
+    private static final float LOADING_VECTOR_MODULE = VectorUtil.dotProduct(new float[257], new float[257]);
```

Comment:
This is not pretty, but necessary. I think the change is ok. If the problem is causing sufficient failures it might be best to push the change as is. However, I would like to revisit this issue with the intention of eventually being in a position to remove this.

Reply:
Could we move this to a static init block, and not retain the computed value?

Reply:
Keeping a reference to the float array does not prevent the module from being "unloaded".

I would just initialize the VectorUtil class and save a class reference in a static final.

Reply:
Actually computing a vector product should not be needed.

Reply:
Unfortunately, computation is needed because the issue is inside the JDK's VectorIntrinsics class, which is not initialised by just loading any API classes. You kinda need to operate on it :-(.

Reply:
Initialising VectorUtil would work if VectorUtilPanamaProvider had something like this:

```java
  static {
    int l = AccessController.doPrivileged((PrivilegedAction<Integer>) () ->
       IntVector.fromArray(IntVector.SPECIES_128, new int[] { 1, 2, 3, 4}, 0).length());
  }
```

Which is what I would propose, if Lucene wants to workaround this too ?